### PR TITLE
CY-866 Add ability to get_capability to get nested property

### PR DESCRIPTION
**Pull request by davidginzbourg**
- This makes it behave similarly to get_attribute.
- Preserves old behavior.

### CODE REVIEW

File: dsl_parser/functions.py [hunk 1]
```diff
@@ -57,6 +57,18 @@ def _is_function(value):
         and list(value.keys())[0] in TEMPLATE_FUNCTIONS
 
 
+def _get_attr_drill_down_string(attr_list):
```
Comment (tyacbovi) [inline dsl_parser/functions.py]: Please add documentation

File: dsl_parser/functions.py [hunk 2]
```diff
@@ -156,8 +168,7 @@ def parse_args(self, args):
             raise ValueError(
                 "Illegal argument(s) passed to {0} function. Expected either"
                 "a string or a list [input_name [, key1/index1 [,...]]] but "
-                "got {1}".format(self.name, ','.join(
-                    [str(arg) for arg in args])))
+                "got {1}".format(self.name, args))
```
Comment (tyacbovi) [inline dsl_parser/functions.py]: Why is the str cast is not necessary anymore?

File: dsl_parser/tests/test_get_capability.py [hunk 3]
```diff
@@ -260,27 +260,9 @@ def test_get_capability_short_list(self):
...
-        type: type
-        properties:
-            property: { get_capability: [ too, many, items ] }
-"""
-        self._assert_parsing_fails(
-            yaml,
-            message="`get_capability` function argument should be a list "
-                    "with 2 elements - the deployment ID and the capability "
-                    "ID. Instead it is: ['too', 'many', 'items']"
+                    "with 2 elements at least - [ deployment ID, capability "
```
Comment (tyacbovi) [inline dsl_parser/tests/test_get_capability.py]: There are missing positive tests and some neg tests

File: dsl_parser/functions.py (hunk 2 above)
Comment (davidginzbourg) [reply dsl_parser/functions.py]: I've tested in `ipython` and the output is basically the same.
```
In [1]: a = [u'one',2,3]

In [2]: str(a)
Out[2]: "[u'one', 2, 3]"

In [3]: ','.join([str(i) for i in a])
Out[3]: 'one,2,3'
```

File: dsl_parser/functions.py [hunk 4]
```diff
@@ -897,3 +908,58 @@ def replace_with_raw_function(*args):
 
     # Change previously replaced get_property instances with raw values
     scan.scan_service_template(plan, replace_with_raw_function, replace=True)
+
+
+def get_nested_capability_attribute(root, path):
+    def _get_error_string(attr_list):
```
Comment (tyacbovi) [inline dsl_parser/functions.py]: It's the same as _get_attr_drill_down_string

File: dsl_parser/functions.py [hunk 5]
```diff
@@ -897,3 +908,58 @@ def replace_with_raw_function(*args):
 
     # Change previously replaced get_property instances with raw values
     scan.scan_service_template(plan, replace_with_raw_function, replace=True)
+
+
+def get_nested_capability_attribute(root, path):
```
Comment (tyacbovi) [inline dsl_parser/functions.py]: not clear name

File: dsl_parser/functions.py (hunk 4 above)
Comment (davidginzbourg) [reply dsl_parser/functions.py]: Oops.

File: dsl_parser/functions.py [hunk 6]
```diff
@@ -57,6 +57,27 @@ def _is_function(value):
         and list(value.keys())[0] in TEMPLATE_FUNCTIONS
 
 
+def _get_attr_drill_down_string(attr_list):
+    """This basically returns a string that represents how the attribute drill
```
[the remark this answers is not in the document]
Comment (davidginzbourg) [reply dsl_parser/functions.py]: Edited.

File: dsl_parser/functions.py (hunk 5 above)
Comment (davidginzbourg) [reply dsl_parser/functions.py]: Changed.

File: dsl_parser/functions.py [hunk 7]
```diff
@@ -58,6 +58,25 @@ def _is_function(value):
         and list(value.keys())[0] in TEMPLATE_FUNCTIONS
 
 
+def _get_attr_drill_down_string(attr_list):
```
[the remark this answers is not in the document]
Comment (davidginzbourg) [reply dsl_parser/functions.py]: How about now?

**Pull request merged**